include/seqcmd.h

```diff
@@ -0,0 +1,652 @@
+    /* 0x5 */ SEQ_CMD_SET_PLAYER_FREQ,
+    /* 0x6 */ SEQ_CMD_SET_CHANNEL_VOL,
+    /* 0x7 */ SEQ_CMD_SET_PLAYER_IO,
+    /* 0x8 */ SEQ_CMD_SET_CHANNEL_IO,
+    /* 0x9 */ SEQ_CMD_DISABLE_CHANNEL_IO,
+    /* 0xA */ SEQ_CMD_DISABLE_CHANNELS,
+    /* 0xB */ SEQ_CMD_TEMPO_CMD,
+    /* 0xC */ SEQ_CMD_SETUP_CMD,
+    /* 0xD */ SEQ_CMD_SET_CHANNEL_FREQ,
+    /* 0xE */ SEQ_CMD_GLOBAL_CMD,
+    /* 0xF */ SEQ_CMD_RESET_HEAP
+} SeqCmdType;
+
+// ==== Secondary commands ====
+
+// Subset of SEQ_CMD_TEMPO_CMD
+typedef enum {
+    /* 0x0 */ SEQ_SUB_CMD_TEMPO_SET,
+    /* 0x1 */ SEQ_SUB_CMD_TEMPO_SPEED_UP,
+    /* 0x2 */ SEQ_SUB_CMD_TEMPO_SLOW_DOWN,
+    /* 0x3 */ SEQ_SUB_CMD_TEMPO_SCALE,
+    /* 0x4 */ SEQ_SUB_CMD_TEMPO_RESET
+} TempoCmdType;
+
+// Subset of SEQ_CMD_SETUP_CMD
+typedef enum {
+    /* 0x0 */ SEQ_SUB_CMD_SETUP_RESTORE_VOLUME,
+    /* 0x1 */ SEQ_SUB_CMD_SETUP_SEQ_UNQUEUE,
+    /* 0x2 */ SEQ_SUB_CMD_SETUP_RESTART_SEQ,
+    /* 0x3 */ SEQ_SUB_CMD_SETUP_TEMPO_SCALE,
+    /* 0x4 */ SEQ_SUB_CMD_SETUP_TEMPO_RESET,
+    /* 0x5 */ SEQ_SUB_CMD_SETUP_PLAY_SEQ,
+    /* 0x6 */ SEQ_SUB_CMD_SETUP_SET_FADE_TIMER,
+    /* 0x7 */ SEQ_SUB_CMD_SETUP_RESTORE_VOLUME_IF_QUEUED,
+    /* 0x8 */ SEQ_SUB_CMD_SETUP_RESTORE_VOLUME_WITH_SCALE,
+    /* 0x9 */ SEQ_SUB_CMD_SETUP_DISABLE_CHANNELS,
+    /* 0xA */ SEQ_SUB_CMD_SETUP_SET_PLAYER_FREQ,
+    /* 0xE */ SEQ_SUB_CMD_SETUP_POP_CACHE = 0xE,
+    /* 0xF */ SEQ_SUB_CMD_SETUP_RESET_SETUP_CMDS
+} SetupCmdType;
+
+// Subset of SEQ_CMD_GLOBAL_CMD
+typedef enum {
+    /* 0x0 */ SEQ_SUB_CMD_GLOBAL_SET_SOUND_MODE,
+    /* 0x1 */ SEQ_SUB_CMD_GLOBAL_DISABLE_NEW_SEQUENCES
+} SubCmdType;
+
+// ==== Audio Sequence Primary Commands ====
+
+/**
+ * ARGS
+ *   playerIndex (p), fadeInTimer (t), seqArgs (a), seqId (s)
+ *
+ * FORMAT
+ *   Capital U is unused
+ *   0pttaass
+ *
+ * DESCRIPTION
+ *   Request a sequence (seqId) to be played on the specified player (playerIndex).
+ *   The sequence will gradually fade in over the course of (fadeInTimer).
+ *   How (fadeInTimer) is interpreted depends on (seqArgs)
+ *
+ *   seqArgs = 8 and seqArgs = 9 (commonly used arguments) appear to have no functionality, 
+ *   although unused commands may suggest  these were intended to be a an priority for an unimplemented priority system.
+ *   
+ *   ((seqArgs & 0x7F) != 0x7F) will interpret fadeInTimer as number of frames at 30 fps
+ *   ((seqArgs & 0x7F) == 0x7F) will interpret fadeInTimer as skip ticks
+ *   (seqArgs & 0x80) i.e. (seqArgs >= 0x80) has an incomplete implementation
+ *   and is not functional in Ocarina of Time. No sequence will play as a result.
+ *   The code and purpose for seqArgs >= 0x80 was completed and is functional in Majora's Mask.
+ */
+#define AudioSeqCmd_PlaySequence(playerIndex, fadeInTimer, seqArgs, seqId)                           \
+    Audio_QueueSeqCmd((SEQ_CMD_PLAY << 28) | ((u8)(playerIndex) << 24) | ((u8)(fadeInTimer) << 16) | \
+                      ((u8)(seqArgs) << 8) | (u16)(seqId))
```

Comment:
I can see why it was done this way, but we should double check that we are ok with introducing macros that are stylized as functions

I know they basically behave as one, but it also can be a bit misleading compared to how with handle macros throughout the rest of the codebase

Reply:
These macro's get quite messy and long once it's capitalized and switched to snake case, but I agree that not doing that breaks away from the style of decomp. I do think it's worth discussion, I'll make another branch with the typical macro style to compare

Reply:
For comparison:
https://github.com/engineer124/oot/compare/audio_seqCmd...engineer124:audio_seqCmd_Macro?expand=1

Reply:
didnt scroll through it all, but it doesnt seem too bad tbh
I could also see the prefix just being SEQCMD, it identifies it enough for what it is i think. but can see the argument for keeping it

Reply:
Right, I've been using `SEQ_` instead of `AUDIO_SEQ_` in other PRs and that's been fine, so I think `SEQCMD_` should be fine as well. Yeah the macros were less terrible than I thought they would be, I'll let it sit for a bit but yeah it might be fine keeping to decomps style.

Just thinking ahead past this PR though, that does raise the question on capitalization for all 4 sets of audio commands. These are the first of 4 sets of audio commands to be fully documented (these commands are interpreted in `audio_seqcmd.c`). There's also the:
2) `Audio_QueueCmd` -> `AudioInternalCmd_XXX` or maybe even just `AudioCmd_XXX` if we go with `SeqCmd_`, which those commands are processed in `code_800E4FE0.c` and are used to transfer requests from the graph thread to the audio thread.
3) `Seq_` commands for the midi-based script WIP which are processed in `audio_seqplayer.c`, see here what those macros look like:
https://github.com/MNGoldenEagle/oot/blob/2d7034dcdbcef6558b4a32dd3372b5515beec9a5/src/audio/000%20Sound%20Effects.mus#L10
4) Audio rsp commands (ex. `aFilter`) in `abi.h` which are processed in `audio_synthesis.c`

Reply:
(just clarifying that I think the functions make sense to have Audio as a prefix, but that a macro that warps the functions probably doesnt need them. So `SEQCMD_` would prob work for a macro that wraps `Audio_QueueSeqCmd`

But yea thinking ahead is fair, im not sure what would we best just yet.
For the rsp commands at least, I think `aMacroName` is a fine exception to make to be able to parallel `gSPMacroName` from gbi

Reply:
if they really are for the rsp specifically, maybe `aSPMacroName`

Reply:
I've gone and changed these macros to the capitalized `SEQCMD_` style